src/main/java/org/sbx/factories/SiteFactory.java

```diff
@@ -0,0 +1,21 @@
+package org.sbx.factories;
+
+import org.sbx.abstracts.Site;
+import org.sbx.objects.Door;
+import org.sbx.objects.Wall;
+
+/**
+ * Created by aloginov on 27.09.16.
+ */
+public class SiteFactory {
+
+    public Site getSite(String siteName){
+        if (siteName.equals("room"))
```

Comment:
You don't need a siteName to distinguish an object class. Use embedded Java methods